src/App.test.tsx

```diff
@@ -1,9 +1,91 @@
-import React from 'react';
-import { render, screen } from '@testing-library/react';
-import App from './App';
-
-test('renders learn react link', () => {
-  render(<App />);
-  const linkElement = screen.getByText(/learn react/i);
-  expect(linkElement).toBeInTheDocument();
+import React from "react";
+import { fireEvent, render, screen, act } from "@testing-library/react";
+import { App } from "./App";
+import { CardProps } from "./component/Card";
+
+jest.mock("./component/Card", () => ({
+  Card: ({ card, handleCardClick }: CardProps) => (
+    <div onClick={() => handleCardClick(card)}>Card</div>
+  ),
+}));
+
+describe("App", () => {
+  it("renders title,subtitle and button", async () => {
+    render(<App />);
+    const title = screen.getByTestId("title");
+    const subtitle = screen.getByTestId("sub-title");
+    expect(title).toBeInTheDocument();
+    expect(subtitle).toBeInTheDocument();
+
+    const btnEasy = screen.getByTestId("btn-easy");
+    const btnMedium = screen.getByTestId("btn-medium");
+    const btnHard = screen.getByTestId("btn-hard");
+
+    expect(btnEasy).toBeInTheDocument();
+    expect(btnMedium).toBeInTheDocument();
+    expect(btnHard).toBeInTheDocument();
+  });
+
+  it("displays 8 cards when clicked on easy", async () => {
```

Comment:
```suggestion
  it("Should display 8 cards when clicked on easy", async () => {
```

Please make this changes for all the test titles.